src/modules/verifiable-credentials/verifiable-credentials.types.ts

```diff
@@ -0,0 +1,97 @@
+import { TypedDataField } from '@ethersproject/abstract-signer';
+import {
+  ICredentialSubject,
+  IJsonLdCredential,
+  IPresentation,
+  IPresentationDefinition,
+} from '@sphereon/pex';
+
+export interface IssuerFields {
+  key: string;
+  value: string | number;
+}
+export interface CreateCredentialParams {
+  subject: string;
+  namespace: string;
+  version: string;
+  issuerFields?: IssuerFields[];
+}
+export interface Credential<T extends ICredentialSubject>
+  extends IJsonLdCredential {
+  credentialSubject: T;
+}
+
+export interface EWFCredentialSubject extends ICredentialSubject {
```

Comment:
Better to call it `RoleCredentialSubject`. EWF can be about more than just roles 😄 
```suggestion
export interface RoleCredentialSubject extends ICredentialSubject {
```